auction-server/src/auction.rs

```diff
@@ -915,30 +918,88 @@ pub fn verify_submit_bid_instruction_svm(
-        tracing::error!(
-            "Account position not found in instruction: {:?} - {}",
-            instruction,
-            position,
-        );
-        RestError::BadParameters("Account not found in submit_bid instruction".to_string())
-    })?;
+    let static_accounts = tx.message.static_account_keys();
+    let tx_lookup_tables = tx.message.address_table_lookups();
+
+    let account_position = submit_bid_instruction
+        .accounts
+        .get(position)
+        .ok_or_else(|| {
+            RestError::BadParameters("Account not found in submit_bid instruction".to_string())
+        })?;
 
     let account_position: usize = (*account_position).into();
-    let account = accounts.get(account_position).ok_or_else(|| {
-        tracing::error!(
-            "Account not found in transaction accounts: {:?} - {}",
-            accounts,
-            account_position,
-        );
-        RestError::BadParameters("Account not found in transaction accounts".to_string())
-    })?;
+    if let Some(account) = static_accounts.get(account_position) {
+        return Ok(*account);
+    }
+
+    match tx_lookup_tables {
+        Some(tx_lookup_tables) => {
+            let lookup_accounts: Vec<(Pubkey, u8)> = tx_lookup_tables
+                .iter()
+                .flat_map(|x| {
+                    x.writable_indexes
+                        .clone()
+                        .into_iter()
+                        .map(|y| (x.account_key, y))
+                })
+                .chain(tx_lookup_tables.iter().flat_map(|x| {
+                    x.readonly_indexes
+                        .clone()
+                        .into_iter()
+                        .map(|y| (x.account_key, y))
+                }))
+                .collect();
+
+            let account_position_lookups = account_position - static_accounts.len();
+            find_and_query_lookup_table(lookup_accounts, account_position_lookups, client).await
+        }
+        None => Err(RestError::BadParameters(
+            "No lookup tables found in submit_bid instruction".to_string(),
+        )),
+    }
+}
 
+async fn find_and_query_lookup_table(
+    lookup_accounts: Vec<(Pubkey, u8)>,
+    account_position: usize,
+    client: &RpcClient,
+) -> Result<Pubkey, RestError> {
+    let (table_to_query, index_to_query) =
+        lookup_accounts.get(account_position).ok_or_else(|| {
+            RestError::BadParameters("Lookup table not found in lookup accounts".to_string())
+        })?;
+
+    query_lookup_table(table_to_query, *index_to_query as usize, client).await
+}
+
+async fn query_lookup_table(
+    table: &Pubkey,
+    index: usize,
+    client: &RpcClient,
+) -> Result<Pubkey, RestError> {
+    let table_data = client
+        .get_account_with_commitment(table, CommitmentConfig::processed())
+        .await
+        .map_err(|_e| {
+            RestError::BadParameters("Failed getting lookup table account data".to_string())
```

Comment:
This is a server error since we were not able to make the RPC call. so something like `TemporarilyUnavailable` makes more sense